recirq/quantum_chess/quantum_board.py

```diff
@@ -133,6 +143,12 @@ def undo_last_move(self) -> bool:
                 return False
         return True
 
+    def record_time(self, t0, t1: Optional[float] = None) -> None:
+        if t1 is None:
+            t1 = time.perf_counter()
+        self.debug_log += (f"sample_with_ancilla takes {t1 - t0:0.4f} seconds.\n")
```

Comment:
I wonder if we should make "sample_with_ancilla" as a parameter. Do you think we will later want to record other calls?

Reply:
Thanks Doug. Now it becomes a parameter "action".